Rust_Scylla_Driver/notes/restserver/src/main.rs

```diff
@@ -0,0 +1,243 @@
+    while let Some(note) = jiter.next() {
+        ret.push(web::Json(note));
+    }
+    Ok(ret)
+}
+
+#[get("/note/{id}")]
+async fn get_by_id(
+    path: web::Path<String>,
+    state: Data<AppState>,
+) -> Result<HttpResponse, Error> {
+    let now = Instant::now();
+    let id = path.into_inner();
+    info!("get_by_id {}", id);
+
+    let ret = get_note(&state.db_svc, &id).await?;
+
+    let elapsed = now.elapsed();
+    info!("get_by_id time: {:.2?}", elapsed);
+    Ok(HttpResponse::Ok().json(ret))
+}
+
+#[delete("/note/{id}")]
+async fn delete_by_id(
+    path: web::Path<String>,
+    state: Data<AppState>,
+) -> Result<HttpResponse, Error> {
+    let now = Instant::now();
+    let id = path.into_inner();
+    info!("get_by_id {}", id);
+
+    let ret = delete_note(&state.db_svc, &id).await?;
+
+    let elapsed = now.elapsed();
+    info!("delete_by_id time: {:.2?}", elapsed);
+    if !ret.is_none() {
+        Err(ErrorInternalServerError(ret.unwrap().to_string()))
+    } else {
+    Ok(HttpResponse::Ok().json(r#"{ "status": "OK"}"#)) }
+}
+
+async fn delete_note(
+    db: &ScyllaDbService,
+    id: &str,
+) -> Result<Option<QueryError>, Error> {
+    let ret = db
+        .delete_note(id)
+        .await
+        .map_err(ErrorInternalServerError);
+    ret
+}
+
+async fn get_note(
+    db: &ScyllaDbService,
+    id: &str,
+) -> Result<Json<Option<Note>>, Error> {
+    let db_notes = db
+        .get_note(id)
+        .await
+        .map_err(ErrorInternalServerError)?;
+
+    let note = Note::from(db_notes);
+
+    Ok(web::Json(note))
+}
+
+#[post("/note")]
+async fn insert_note(
+    payload: web::Json<AddNoteRequest>,
+    state: Data<AppState>,
+) -> Result<HttpResponse, Error> {
+    info!("Ingest Request: {:?}", payload.topic);
+    let now = Instant::now();
+
+    let mut handlers: Vec<JoinHandle<_>> = Vec::new(); //sample multi insert parsing
+
+    let jid = payload.id.clone().unwrap_or_default();
+    let permit = state.semaphore.clone().acquire_owned().await;
+    let id = if jid!="" {Uuid::parse_str(jid.as_str()).unwrap()} else { Uuid::new_v4() };
+    handlers.push(task::spawn(add_note(
+        id,
+        state.clone(),
+        payload.topic.to_string(),
+        payload.content.to_string(),
+        permit,
+    )));
+
+    debug!("Waiting for notes to be processed...");
+    for thread in handlers {
+        match thread.await {
+            Err(e) => return Err(ErrorInternalServerError(e)),
+            Ok(r) => {
+                if let Err(e) = r {
+                    error!("Error: {:?}", e);
+                    return Err(ErrorInternalServerError(e));
+                }
+            }
+        }
+    }
+
+    let elapsed = now.elapsed();
+    info!("Ingestion Time: {:.2?}", elapsed);
+    Ok(HttpResponse::Ok().json(r#"{ "status": "OK"}"#))
+}
+
+async fn add_note(
+    id: Uuid,
+    state: Data<AppState>,
+    topic: String,
+    content: String,
+    permit: Result<OwnedSemaphorePermit, AcquireError>,
```

Comment:
It seems that `permit` is not used at all.

Reply:
Thanks @Gor027 , @tarzanek WDYT?

Reply:
yes, the goal was to only allow certain number to get into this function 
using 
    let permit = state.semaphore.clone().acquire_owned().await;

I don't know if compiler wouldn't optimize above away if it wouldn't be passed as object
example of semaphore api shows it should be just passed on to critical section (yes, silly way, wondering if rust has something better)